### ISSUE TITLE
Swift Package Manager support

### ISSUE THREAD

**Issue by vi4m** [bow-swift/nef#33]
Thanks for the awesome tool!
It would be great to have SPM support!

**Comment by miguelangel-dev** [comment]
After Apple's announcement, I think we should try to migrate from `cocoapods` to SPM. This issue will need:
1. [nef playground] create an initial playground, compatible with third-party libraries from SPM, or unless with an own solution.
2. [nef compile] SPM will be the default dependency manager.
3. [nef compile --podfile] we need to keep compatibility with `cocoapods`: for nef compiler, we can decide use cocoapods with the flat `--podfile`

After completing this issue, homebrew only will have one dependency, `Xcode` and `macos` (the only place where Xcode playgrounds are compatible for now)

**Comment by miguelangel-dev** [comment]
We have started the path to add compatibility to other dependency managers:
- Carthage (**done in** #89 #90 #91 - [released in **version 0.4**](https://github.com/bow-swift/nef/releases/tag/0.4.0))
- SPM (**pending** Apple fixes the SPM bug in Xcode - 47668990)

Swift Package Manager is not as easy as we can expect. Using the new Xcode 11 and SPM integrated does not resolve dependencies for a nested Playground as it does in CocoaPods or Carthage (Apple introduced it in [Xcode 9](https://help.apple.com/xcode/mac/9.0/#/devc9b33111c)). 

We will start to decouple CocoaPods from nef, and integrate Carthage when it solves (or not) we will see how to integrate SPM in `nef`.

**more information**
https://developer.apple.com/documentation/xcode_release_notes/xcode_11_beta_5_release_notes
```
Xcode 11 beta 5
Known Issues
...
Playgrounds in a workspace can’t import targets from a Swift package. (47668990)
...
```
---
https://developer.apple.com/documentation/xcode_release_notes/xcode_11_beta_6_release_notes
```
Xcode 11 beta 6

Playgrounds
Known Issues
...
Playgrounds in a workspace can’t import targets from a Swift package. (47668990)
...
```
---
https://developer.apple.com/documentation/xcode_release_notes/xcode_11_release_notes
```
Xcode 11 GM - bug #47668990 not fixed
```

**Comment by twof** [comment]
Is support possible for SPM now? In the release notes for Xcode 11.3, bug #47668990 is no longer listed.

https://developer.apple.com/documentation/xcode_release_notes/xcode_11_3_release_notes

**Comment by truizlop** [comment]
Thanks for letting us know! We'll check and if everything is working, we will provide a new release supporting it.

**Comment by miguelangel-dev** [comment]
Thanks @twof,

I have checked it and unfortunately, it is not solved yet. 

Apple usually adds the "**known issues**" when they detect a bug, and in future releases, they do not list it again; just when they fix the bug it is announced in the section "**Resolved Issues**"

So, for now, we have to wait, let's see during the Q1-20 they solve it ^^ - We will keep you updated in this thread!

**Comment by twof** [comment]
@miguelangel-dev @truizlop Thanks!

**Comment by miguelangel-dev** [comment]
One year later, Apple has updated SPM and in the new Xcode 12 releases notes, we can find:
```
New Features
Xcode Playgrounds can now import and use Swift packages and frameworks. Select the Build Active Scheme checkbox in the playground’s File inspector and ensure that the active scheme builds the package or framework target. (60612584)

Xcode’s Report Navigator now includes Playground build logs. (56351128)

Xcode Playgrounds now build the active scheme’s targets and make them importable when Build Active Scheme is enabled in the File inspector. (60089627)
```

It is in our roadmap, and we will work in add SPM compatibility to nef Playgrounds. On this way, we can take advantage of the whole nef features even with swift-package-manager. 🎉

Source: https://developer.apple.com/documentation/xcode-release-notes/xcode-12-beta-release-notes/

**Comment by miguelangel-dev** [comment]
It will be available in `nef 0.7`. Thanks to everyone for the support and interest :)
